provider/aws/cloudformation.go

```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/ecr"
```

Comment:
don't like adding so much of this to the cloudformation specific code...cohesion is too low

Reply:
I need the ./ecr package to delete the ECR repo that always gets left behind.  I could have sworn there was already an issue file for this, but there isn't.  Anyway, I know it has been mentioned at least internally (maybe a Stelligent Slack chat).  Open to alternative suggestions on how to take care of the ECR repo.

Reply:
I'd prefer all the stuff to empty buckets and ECR repos to be moved out of this file and keep it specific to the purge function.  makes me nervous to leak some of the purge functionality into this highly shared location.